src/systems/input_system.rs

```diff
@@ -92,7 +92,8 @@ impl<'a> System<'a> for InputSystem {
                         KeyCode::Right => renderable.position.x += 1,
                         _ => ()
                     }
-                    if let Some(directional) = directionals.get_mut(entities.entity(id)) {
+                    if player.contains(entities.entity(id)) {
```

Comment:
player will definitely change direction, do not need to check if player.contains